OracleTuxedo/core/samples/ws_ssl_svr/ws_svr_runme.sh

```diff
@@ -0,0 +1,166 @@
+#!/bin/bash
+#
+# Copyright (c) 2022 Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+# 
+# Shell script to build and run ws_svr.  It assume TUXDIR has been set, or that 
+# Tuxedo has been installed to: ~/tuxHome/tuxedo12.1.3.0.0   If not, invoke 
+# this script with a single argument indicating the location of TUXDIR.
+#
+# Usage: source ws_svr_runme.sh
+#
+mkdir -p /u01/oracle/user_projects
+cd /u01/oracle/user_projects || return 1
+rm -rf ws_svr
+mkdir ws_svr
+cd ws_svr || return 1
+
+# Create environment setup script setenv.sh
+HOSTNAME=$(uname -n)
+APPDIR=$(pwd)
+export HOSTNAME APPDIR
+export WS_PORT=9055
+export SSL_PORT=9060
+#export TUX_LOADBAL_IP="138.3.79.82"
+export SEC_PRINCIPAL_NAME="ISH_tuxqa"            # ssl
+
+cat >setenv.sh << EndOfFile
+source ${TUXDIR}/tux.env
+export HOSTNAME=${HOSTNAME}
+export APPDIR=${APPDIR}
+export TUXCONFIG=${APPDIR}/tuxconfig
+export TM_SECURITY_CONFIG=NONE
+export TM_ALLOW_NOTLS=Y
+export SSL_PORT=${SSL_PORT}                      # ssl
+export ULOG_SSLINFO=yes                          # ssl
+export TM_MIN_PUB_KEY_LENGTH=1024                # ssl
+export SEC_PRINCIPAL_NAME=${SEC_PRINCIPAL_NAME}  # ssl
+export PASSVAR=tuxsvr123                         # ssl
+export IPCKEY=112233
+export NLSPORT=12233
+export JMXPORT=22233
+export WSNADDR="//${HOSTNAME}:${WS_PORT}"
+unset NATADDR_OPT
+if [[ -n "${TUX_LOADBAL_IP}" ]]; then
+    export NATADDR_OPT=" -H //${TUX_LOADBAL_IP}:${WS_PORT} "
+fi
+EndOfFile
+
+# shellcheck disable=SC1091
+source ./setenv.sh
+
+
+
+# create ssl certificates and PKCS12 wallet
+WALLET_DIR=${APPDIR}/wallet/wallet.${SEC_PRINCIPAL_NAME}
+mkdir -p "${WALLET_DIR}"
+mkdir -p "${APPDIR}"/CA
+
+    cd "${APPDIR}"/CA || return 1
+    mkdir certs newcerts private
+    cp -p  /u01/oracle/openssl.cnf .
+    touch index.txt 
+    touch certindex.txt
+    echo "0100" > serial
+
+    # create CA key and certificate
+    openssl req -new -x509 -extensions v3_ca -keyout private/ca_key.pem -out certs/ca_cert.pem \
+                -days 3650 -config ./openssl.cnf  \
+                -passout pass:'ca123'  \
```

Comment:
You are not permitted to set hard-coded passwords in any way. Please replace this with a script that randomly generates a string during the build process. The script should output that random string only during the build process.

Reply:
You should also allow the user to provide their own password in a secure manner.